Code/Framework/AzCore/AzCore/DOM/Backends/JSON/JsonSerializationUtils.cpp

```diff
@@ -0,0 +1,675 @@
+            }
+
+            if (m_entryStack.front().m_entryCount != attributeCount)
+            {
+                return FormatVisitorFailure(
+                    VisitorErrorCode::InternalError, "EndObject: Expected %lu attributes but received %lu attributes instead",
+                    attributeCount, m_entryStack.front().m_entryCount);
+            }
+
+            m_entryStack.pop_front();
+            return FinishWrite();
+        }
+
+        Result Key(AZ::Name key) override
+        {
+            return RawKey(key.GetStringView(), Lifetime::Persistent);
+        }
+
+        Result RawKey(AZStd::string_view key, Lifetime lifetime) override
+        {
+            AZ_Assert(!m_entryStack.empty(), "Attempmted to push a key with no object");
+            AZ_Assert(m_entryStack.front().m_isObject, "Attempted to push a key to an array");
+            if (lifetime == Lifetime::Persistent)
+            {
+                m_entryStack.front().m_key.SetString(key.data(), static_cast<rapidjson::SizeType>(key.size()));
+            }
+            else
+            {
+                m_entryStack.front().m_key.SetString(key.data(), static_cast<rapidjson::SizeType>(key.size()), m_result.GetAllocator());
+            }
+            return VisitorSuccess();
+        }
+
+        Result StartArray() override
+        {
+            CurrentValue().SetArray();
+
+            const bool isObject = false;
+            m_entryStack.emplace_front(isObject, CurrentValue());
+            return VisitorSuccess();
+        }
+
+        Result EndArray(AZ::u64 elementCount) override
+        {
+            if (m_entryStack.empty())
+            {
+                return VisitorFailure(VisitorErrorCode::InternalError, "EndArray called without a matching BeginArray call");
+            }
+
+            if (m_entryStack.front().m_isObject)
+            {
+                return VisitorFailure(VisitorErrorCode::InternalError, "Expected EndObject and received EndArray instead");
+            }
+
+            if (m_entryStack.front().m_entryCount != elementCount)
+            {
+                return FormatVisitorFailure(
+                    VisitorErrorCode::InternalError, "EndArray: Expected %lu elements but received %lu elements instead", elementCount,
+                    m_entryStack.front().m_entryCount);
+            }
+
+            m_entryStack.pop_front();
+            return FinishWrite();
+        }
+
+        rapidjson::Document&& TakeDocument()
+        {
+            return AZStd::move(m_result);
+        }
+
+    private:
+        Result FinishWrite()
+        {
+            if (m_entryStack.empty())
+            {
+                return VisitorSuccess();
+            }
+
+            // Retrieve the top value of the stack and replace it with a null value
+            rapidjson::Value value;
+            m_entryStack.front().m_value.Swap(value);
```

Comment:
Heh I gave this a thumbs-up but apparently didn't manage to grok it - do you mean something like:
```
rapidjson::Value nullValue;
rapidjson::Value& value = m_entryStack.front().m_value.Swap(nullValue);
```
I'm actually unclear on how that's faster, but feel like I'm missing something 😅

Reply:
`m_entryStack.front()` is begin called repeatedly, so instead of doing that do something like:
`<insert stored type>& front = m_entryStack.front();` and use `front` instead.